kernel/inductive.ml

```diff
@@ -1209,8 +1209,8 @@ let check_one_fix renv recpos trees def =
               let c = whd_all renv.env c in
               let hd, _ = decompose_app c in
               match kind hd with
-              | Construct _ ->
-                  check_rec_call renv []
+              | Construct _ | CoFix _ ->
+                  check_rec_call renv stack
                     (Term.applist (mkProj(Projection.unfold p,c), l))
               | _ -> Exninfo.iraise exn
```

Comment:
Why not make this explicit so we definitely don't forget any cases?

Reply:
Good question. We could but there are a lot of cases to add. Wondering what others would do themselves.

Reply:
I think that in the kernel at least, it is considered good practice to fully enumerate the constructors of `constr`.

Reply:
I agree, the catch-all bit me once in the guard as well when we introduced primitive projections.

Reply:
OK, done.